src/app/modules/entities/view-entity-details/view-entity-details.component.html

```diff
@@ -0,0 +1,163 @@
+      </mat-accordion>
+      <div class="user-details-card">
+
+        <mat-card>
+          <div class="row" style="padding-bottom: 2%;">
+            <div class="roles-head">
+              <div style="padding-top: 15px;">
+                <mat-panel-title class="c-font">
+                  Entity List - <span class="text-capital"> {{type}}</span>
+                </mat-panel-title>
+              </div>
+
+              <div class="c-search">
+                <mat-form-field class="custom-form-field">
+                  <mat-label>Search for {{type}}</mat-label>
+                  <input [(ngModel)]="search" autocomplete="off" [appDebounceDirective]="search"
+                    (onValueChange)="subEntity()" #searchInput matInput>
+                  <button mat-icon-button matPrefix>
+                    <mat-icon>search</mat-icon>
+                  </button>
+                </mat-form-field>
+              </div>
+
+              <!-- <div *ngIf="editUserDetails?.organisations.length < editUserDetails?.organisationsList.length">
+              <button mat-button class="actions" (click)="addOrganisation(details)" cdkFocusInitial>Add New
+                organisation</button>
+            </div> -->
+            </div>
+          </div>
+
+          <div class="table-container mat-elevation-z8">
+            <app-spinner *ngIf="listing == false"></app-spinner>
+            <mat-table *ngIf="dataSource" [dataSource]="dataSource" matSort>
+              <span *ngFor="let column of columns">
+                <ng-container [matColumnDef]="column.key">
+                  <mat-header-cell style="padding-right: 2%;" *matHeaderCellDef>{{ column.label}}</mat-header-cell>
+                  <span *ngIf="column.type == 'column'">
+                    <mat-cell style="padding-right: 2%;" *matCellDef="let row" class="text-capital">
+                      <span class="{{getItemCssClassByStatus(row[column.key])}}">
+                        {{row[column.key] ? row[column.key]: '-- --'}}
+                      </span>
+                    </mat-cell>
+                  </span>
+                  <span *ngIf="column.type == 'action'">
+                    <mat-cell *matCellDef="let user">
+                      <button (click)="ViewData(user)" matTooltip="View details" class="actions"
+                        mat-button>View</button> &nbsp;
+                    </mat-cell>
+                  </span>
+                </ng-container>
+              </span>
+              <mat-header-row *matHeaderRowDef="displayedColumns;sticky: true" class="p-button">
+              </mat-header-row>
+              <mat-row *matRowDef="let row; columns: displayedColumns;">
+              </mat-row>
+            </mat-table>
+
+            <div class="message" *ngIf="recordCount == 0 || !dataSource">No records found</div>
+            <mat-paginator [length]="recordCount" [showFirstLastButtons]="true" [pageSize]="10"
+              [pageSizeOptions]="[5, 10, 25, 100]">
```

Comment:
use constants to add here

Reply:
can you please explain to get clear idea?